cypress/pageobject/Users/UserCreation.ts

```diff
@@ -1,65 +1,106 @@
+  typeEmail(email: string) {
+    cy.get("#email").click().type(email);
   }
-
-  selectFacility(name: string) {
-    this.typeIntoInputByName("facilities", name);
-    this.selectOptionContainingText(name);
-    cy.get("input[name='facilities'] + button")
-      .find("#dropdown-toggle")
-      .click();
+  typePassword(password: string) {
+    cy.get("#password").click().type(password);
+  }
+  typePhoneNumber(phone_number: string) {
+    cy.get("#phoneNumber").click().type(phone_number);
+  }
+  typeConfirmPassword(password: string) {
+    cy.get("#c_password").click().type(password);
+  }
+  typeQualification(qualification: string) {
+    cy.get("#qualification").click().type(qualification);
+  }
+  typeDoctorExperience(experience: string) {
+    cy.get("#doctor_experience_commenced_on").click().type(experience);
+  }
+  typeDoctorMedicalCouncilRegNo(regNo: string) {
+    cy.get("#doctor_medical_council_registration").click().type(regNo);
+  }
+  typeWeeklyWorkingHours(workingHrs: string) {
+    cy.get("#weekly_working_hours").click().type(workingHrs);
+  }
+  typeAltPhoneNumber(emergencyPhoneNumber: string) {
+    cy.get("#altPhoneNumber").click().type(emergencyPhoneNumber);
+  }
+  typeDateOfBirth(dob: string) {
+    cy.clickAndTypeDate("#date_of_birth", dob);
+  }
+  typeNewUserFirstName(firstName: string) {
+    cy.get("#first_name").click().type(firstName);
+  }
+  typeNewUserLastName(lastName: string) {
+    cy.get("#last_name").click().type(lastName);
+  }
+  typeNewUserPhoneNumber(phoneNumber: string) {
+    cy.get("#phone_number").click().type(phoneNumber);
+  }
+  clearFirstName() {
+    cy.get("#firstName").click().clear();
+  }
+  clearLastName() {
+    cy.get("#lastName").click().clear();
+  }
+  clearPhoneNumber() {
+    cy.get("#phoneNumber").click().clear();
+  }
+  clearAltPhoneNumber() {
+    cy.get("#altPhoneNumber").click().clear();
+  }
+  clearWeeklyWorkingHours() {
+    cy.get("#weekly_working_hours").click().clear();
+  }
+  clearEmail() {
+    cy.get("#email").click().clear();
   }
 
+  selectUserType(role: string) {
+    cy.clickAndSelectOption("#user_type", role);
+  }
   selectHomeFacility(name: string) {
-    this.clickElementById("home_facility");
-    this.selectOptionContainingText(name);
+    cy.clickAndSelectOption("#home_facility", name);
   }
 
-  setInputDate(dateElementId: string, dateValue: string) {
-    cy.clickAndTypeDate("#" + dateElementId, dateValue);
+  selectGender(gender: string) {
+    cy.clickAndSelectOption("#gender", gender);
+  }
+  selectState(state: string) {
+    cy.clickAndSelectOption("#state", state);
+  }
+  selectDistrict(district: string) {
+    cy.clickAndSelectOption("#district", district);
   }
 
-  selectDropdownOption(dropdownId: string, optionText: string) {
-    this.clickElementById(dropdownId);
-    this.selectOptionContainingText(optionText);
+  selectFacility(name: string) {
+    cy.typeAndSelectOption("input[name='facilities']", name);
```

Comment:
```suggestion
    cy.typeAndSelectOption("input[name='facilities']", name);
```

- use the reusable component typeAndMultiSelectOption
- as mentioned in the issue, read the command.ts file to understand the existing components